pkg/ds/farm.go

```diff
@@ -564,6 +565,25 @@ func (dsf *Farm) spawnDaemonSet(
 
 	desiresCh := ds.WatchDesires(quitSpawnCh, updatedCh, deletedCh)
 
+	go func() {
+		aggregateHealth := replication.AggregateHealth(ds.PodID(), *dsf.healthChecker)
```

Comment:
I may have to do this one at a a time. First, a config option to turn on health monitoring. Then, if everything is good, we optimize by passing the health monitors to the replicators.

This is because if we do both, we have to do this weird thing where if health monitoring is enabled we pass the existing health monitor to the replicator, otherwise we construct a new one when the replicator is constructed.